### PR TITLE
Color reduction args

### PR DESCRIPTION
**Pull request by MathiasMehl**
Adds: 
-R 0 to disable colored structural reductions
-R 1 to enable colored structural reductions and apply all rules
-R 2 to enable colored structural reductions and apply rules in a custom sequence
-D to give a timeout for colored structural reductions

### CODE REVIEW

File: include/PetriEngine/Colored/Reduction/ColoredReducer.h [hunk 1]
```diff
@@ -96,6 +96,13 @@ namespace PetriEngine::Colored {
 
             void skipTransition(uint32_t tid);
 
+            std::vector<ReductionRule *> getSpecifiedReductions(std::vector<uint32_t>& reductions) {
```
Comment (NicEastvillage) [inline include/PetriEngine/Colored/Reduction/ColoredReducer.h:99]: Kunne være private? kunne også kaldes `buildApplicationSequence`

File: include/PetriEngine/options.h [hunk 2]
```diff
@@ -62,8 +62,11 @@ struct options_t {
     const char* modelfile = nullptr;
     const char* queryfile = nullptr;
     int enablereduction = 1; // 0 ... disabled,  1 ... aggresive (default), 2 ... k-boundedness preserving, 3 ... selection
+    int enablecolreduction = 1;
     std::vector<uint32_t> reductions{8,2,3,4,5,7,9,6,0,1};
+    std::vector<uint32_t> colreductions{};
     int reductionTimeout = 60;
+    int colReductionTimeout = 60;
```
Comment (NicEastvillage) [inline include/PetriEngine/options.h:69]: Jeg tror colored reductions er hurtigt færdige, da CPNs er små. Hvad tænker du til 30 sekunder?

File: src/PetriEngine/options.cpp [hunk 3]
```diff
@@ -53,6 +53,16 @@ void options_t::print(std::ostream& optionsOut) {
         optionsOut << ",State_Space_Exploration=DISABLED";
     }
 
+    if (enablecolreduction == 0) {
+        optionsOut << ",Colored_Structural_Reduction=DISABLED";
+    } else if (enablecolreduction == 1) {
+        optionsOut << ",Colored_Structural_Reduction=AGGRESSIVE";
```
Comment (NicEastvillage) [inline src/PetriEngine/options.cpp:59]: Ved ikke om 1 er "aggressive" endnu, men der er i hvert fald "alle"

Comment (MathiasMehl) [reply src/PetriEngine/options.cpp]: [6c12db2](https://github.com/DAT10CPN/verifypn/pull/3/commits/6c12db21878533df219fcec295451ca5afbc27a9)

File: include/PetriEngine/options.h (hunk 2 above)
Comment (MathiasMehl) [reply include/PetriEngine/options.h]: [7b9f88c](https://github.com/DAT10CPN/verifypn/pull/3/commits/7b9f88c2291b020376a25196a515eda324e4bb0b)

File: include/PetriEngine/Colored/Reduction/ColoredReducer.h (hunk 1 above)
Comment (MathiasMehl) [reply include/PetriEngine/Colored/Reduction/ColoredReducer.h]: [4aeacfc](https://github.com/DAT10CPN/verifypn/pull/3/commits/4aeacfca0cb7c05646c8c9222fd8abb8aa1fc1a7)

File: include/PetriEngine/Colored/Reduction/ColoredReducer.h [hunk 4]
```diff
@@ -107,6 +106,16 @@ namespace PetriEngine::Colored {
             std::vector<uint32_t> _skippedPlaces;
             std::vector<uint32_t> _skippedTransitions;
 
+            std::vector<ReductionRule *> _usedReductions{};
```
Comment (NicEastvillage) [inline include/PetriEngine/Colored/Reduction/ColoredReducer.h:109]: I don't think you use this

Comment (MathiasMehl) [reply include/PetriEngine/Colored/Reduction/ColoredReducer.h]: Fixed in [d947046](https://github.com/DAT10CPN/verifypn/pull/3/commits/d947046919a2344a634bd2c9dcfa1ffc7b3bd650)

**Review by NicEastvillage — APPROVED** [review]

**Pull request merged**